### ISSUE TITLE
`ptr_arg` lint wrongly suggests replacing `&mut Vec<_>` parameters with `&mut [_]` if the function body uses the `Vec` API

### ISSUE THREAD

**Issue by sdroege** [rust-lang/rust-clippy#8482]
### Summary

The function in question takes mutable `Vec`s as parameters and modifies them.

### Lint Name

ptr_arg

### Reproducer

I tried this code:

```rust
pub fn generate_reexports(
    env: &Env,
    analysis: &analysis::object::Info,
    module_name: &str,
    contents: &mut Vec<String>,
    traits: &mut Vec<String>,
    builders: &mut Vec<String>,
) {
    let mut cfgs: Vec<String> = Vec::new();
    if let Some(cfg) = general::cfg_condition_string(analysis.cfg_condition.as_ref(), false, 0) {
        cfgs.push(cfg);
    }
    if let Some(cfg) = general::version_condition_string(env, None, analysis.version, false, 0) {
        cfgs.push(cfg);
    }
    if let Some(cfg) = general::cfg_deprecated_string(
        env,
        Some(analysis.type_id),
        analysis.deprecated_version,
        false,
        0,
    ) {
        cfgs.push(cfg);
    }

    contents.push("".to_owned());
    contents.extend_from_slice(&cfgs);
    contents.push(format!("mod {};", module_name));
    contents.extend_from_slice(&cfgs);

    contents.push(format!(
        "{} use self::{}::{};",
        analysis.visibility.export_visibility(),
        module_name,
        analysis.name,
    ));

    if analysis.need_generate_trait() {
        for cfg in &cfgs {
            traits.push(format!("\t{}", cfg));
        }
        traits.push(format!(
            "\tpub use super::{}::{};",
            module_name, analysis.trait_name
        ));
    }

    if has_builder_properties(&analysis.builder_properties) {
        for cfg in &cfgs {
            builders.push(format!("\t{}", cfg));
        }
        builders.push(format!(
            "\tpub use super::{}::{}Builder;",
            module_name, analysis.name
        ));
    }
}
```

I saw this happen:

```
warning: writing `&mut Vec` instead of `&mut [_]` involves a new object where a slice will do
   --> src/codegen/object.rs:589:13
    |
589 |     traits: &mut Vec<String>,
    |             ^^^^^^^^^^^^^^^^ help: change this to: `&mut [String]`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#ptr_arg

warning: writing `&mut Vec` instead of `&mut [_]` involves a new object where a slice will do
   --> src/codegen/object.rs:590:15
    |
590 |     builders: &mut Vec<String>,
    |               ^^^^^^^^^^^^^^^^ help: change this to: `&mut [String]`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#ptr_arg
```

I expected to see this happen: Lint doesn't trigger

### Version

```text
rustc 1.60.0-beta.1 (0a4f984a8 2022-02-22)
binary: rustc
commit-hash: 0a4f984a87c7ba6c74ec3e78442fec955a419e32
commit-date: 2022-02-22
host: x86_64-unknown-linux-gnu
release: 1.60.0-beta.1
LLVM version: 14.0.0
```

### Additional Labels

@rustbot label +I-suggestion-causes-error

**Comment by Jarcho** [comment]
This might be fixed by #8464. Looks like it's the same bug.

**Comment by Jarcho** [comment]
Just checked on the playground and as far as I can tell this no longer occurs.

**Comment by tomhoule** [comment]
I am observing this issue in multiple files in a large project, [here](https://github.com/prisma/prisma-engines/blob/ba8e3c2c2615e6af9303b3930e7de9377999a0e7/migration-engine/connectors/sql-migration-connector/src/sql_renderer/postgres_renderer.rs#L511) for example. We push to that Vec in the function body, so the suggestion is wrong.

Clippy version: `clippy 0.1.60 (7737e0b 2022-04-04)`

**Comment by Jarcho** [comment]
I don't think the fix made it into `1.60`

**Comment by Jarcho** [comment]
This should be fixed in `1.61`

**Comment by kraktus** [comment]
Can be closed then?
